tests/_internal/concurrency/test_cancellation.py

```diff
@@ -35,9 +35,39 @@ def mock_alarm_signal_handler():
         signal.signal(signal.SIGALRM, _previous_alarm_handler)
 
 
-@pytest.mark.parametrize(
-    "cls", [AlarmCancelScope, WatcherThreadCancelScope, AsyncCancelScope]
-)
+async def test_alarm_cancel_scope_repr():
```

Comment:
are we using the `mock_alarm_signal_handler` at lines 25-35? I'm confused as to why using that fixture would not allow us to appropriately mock and handle/skip signals not on the main thread.